docs_src/design/ucr/0005-Hybrid-App-Device-Services.md

```diff
@@ -0,0 +1,179 @@
+and rely on device profiles to describe their handling of data from either south-bound devices or 
+service-based devices.
+
+
+### Motivation
+The EdgeX docs describe a nice, clean framework wherein there are south-bound Device Services that communicate with
+external devices and import their data, and north-bound Application Services that transform that
+device data for external actors; there are separate SDKs for Device and Application Services, and
+there are separate sets of APIs associated with Device and Application Services. In this view, the work of 
+Application Services is triggered by internal EdgeX events (API calls or Message Bus events).
+
+In our implementations, we find several problems with clean separation of Device and Application Services as we 
+design microservices for our industrial products:
+- Our north-bound microservices respond to external, as well as internal, stimuli, and need to maintain
+persistent connections to external actors or to listen persistently for external requests.
+- For analytic or north-bound services managing large amounts of data from south-bound devices, the scale does 
+not permit the use of per-resource rules to describe how each resource is handled; instead, it is necessary to 
+code the rules and let the bulk or batch handling be described by something just like device profiles. 
+These are the 95-98% case, where the coded rules are relatively simple (eg, for interval trending or alarm 
+monitoring); for the remainder, the complex rule cases, the existing EdgeX means of rule handling (triggers and 
+eKuiper rules) are a good solution.
+- Static configuration of analytic and north-bound services with respect to the device data that they must handle
+has proven insufficient; instead, these services need something just like the dynamic provisioning of Device Services.
+- For our analytic, utility, and north-bound services, we almost always choose to add (Device) Resources as the 
+means to manage their configuration, transforms, and status reporting. These Resources are often needed on a 
+per-Device basis (rather than just overall service configuration or status), sometimes extending (adding to) the 
+data of the original south-bound devices.
+- The lack of a C/C++ Application Services SDK, for developers not familiar with Go and for north-bound protocol 
+libraries which lack a Go implementation.
+
+While we could choose to address some of these points by creating specific APIs for each service to configure
+and get status, it seems that much of development would be spent in creating and maintaining these bespoke APIs
+which, in the end, reduce to Getting and Setting internal resource points inside the service. 
+So the simpler approach is to present them as Device Resources to be managed with the well-known core-command
+GET and PUT commands, provided that these Resources are well-described in their Device Profiles.
```

Comment:
What I get here is the need to have the same commanding capability for the resources added. To me this sounds like more of a Device Service analytic extension service which builds on the existing device services? Rather than receiving data from the device it receives the data from the device services, extends it and publishes it.

Reply:
Yes, that sounds correct. I think we agree, except maybe that as north-bound or at least north-side services, they would be seen as "Application Services" by a general audience, per the EdgeX-Docs descriptions.

Reply:
Yep